File: SaneScanner/UIImage+SY.m
//
//  UIImage+SY.m
//  SaneScanner
//
//  Created by Stan Chevallier on 04/03/2016.
//  Copyright © 2016 Syan. All rights reserved.
//

#import "UIImage+SY.h"
#import <UIImage+SYKit.h>
#import "SYSaneScanParameters.h"
#import <ImageIO/ImageIO.h>
#import "SYTools.h"

@implementation UIImage (SY)

+ (UIImage *)sy_addIconWithColor:(UIColor *)color
{
    // iOS defaults @2x for UIBarButtonItemSystemAdd
    return [self sy_addIconWithColor:color
                                size:35.
                         borderWidth:3.];
}

+ (UIImage *)sy_addIconWithColor:(UIColor *)color
                            size:(CGFloat)size
                     borderWidth:(CGFloat)borderWidth
{
    CGRect verticalRect   = CGRectMake((size - borderWidth)/2., 0, borderWidth, size);
    CGRect horizontalRect = CGRectMake(0, (size - borderWidth)/2., size, borderWidth);
    CGRect wholeRect      = CGRectMake(0, 0, size, size);
    
    UIGraphicsBeginImageContextWithOptions(wholeRect.size, NO, 0.0);
    
    [color setFill];
    
    UIBezierPath *path = [UIBezierPath bezierPathWithRect:verticalRect];
    [path appendPath:[UIBezierPath bezierPathWithRect:horizontalRect]];
    [path fill];
    
    UIImage *img = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return img;
}

@end

File: SaneScanner/UIImage+SY.swift
//
//  UIImage+SY.swift
//  SaneScanner
//
//  Created by Stanislas Chevallier on 06/03/2019.
//  Copyright © 2019 Syan. All rights reserved.
//

import UIKit

extension UIImage {
    static func testImage(size: CGFloat) -> UIImage? {
        let letter = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789".randomElement() ?? "A"
        let label = UILabel()
        label.font = UIFont.boldSystemFont(ofSize: size / 2)
        label.textAlignment = .center
        label.text = String(letter)
        label.textColor = .white
        label.backgroundColor = .clear
        label.frame = CGRect(x: 0, y: 0, width: size, height: size)
        
        UIGraphicsBeginImageContextWithOptions(CGSize(width: size, height: size), true, 1)
        defer { UIGraphicsEndImageContext() }
        
        UIColor.random.setFill()
        UIBezierPath(rect: label.bounds).fill()
        
        label.draw(label.bounds)
        return UIGraphicsGetImageFromCurrentImageContext()
    }
}
